lib/queue.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+class Queue {
+  constructor() {
+    this.elements = [];
+    this.offset = 0;
+  }
+
+  push(...element) {
+    this.elements.push(...element);
+    return this;
+  }
+
+  shift() {
```

Comment:
It may be better to use enqueue/dequeue methods and under the hood add elements with array.unshift, and  remove elements with offset/slice. Need to check performance and compare

Reply:
This approach was to match current array methods (push/shift) to minimize refactoring, but I agree that enqueue/dequeue could be more convinient.

Reply:
Also here is `test/queue-performance.js` performance comparison test to compare queues of 100k elements